### ISSUE TITLE
wsk "-a" accepts invalid JSON and turns it into a string annotation

### ISSUE THREAD

**Issue by starpit** [openwhisk/openwhisk#137]
we just merged in a breaking change to installWatson.sh:

```
install 'watson/languageId.js'      watson/languageId \
    -a description 'Identify language' \
    -a parameters '[ {"name":"username", "required":true, "bindTime":true}, {"name":"password", "required":true, "type":"password", "bindTime":t\
rue}, {"name":"payload", "required":true} ]' \
    -a sampleInput '{"payload": "Bonjour", "username":"XXX", "password":"XXX"}' \
    -a sampleOutput '{"language": "French", confidence: 1}'
```

the sampleOutput annotation is missing the "quotes" around confidence. wsk turns this into a string annotation, which was not what the author intended.

Option 1: change the meaning of -a to mean "strict JSON", i.e. the default will be to fail if the annotation is not parseable as JSON; add a --nonstrict top-level switch, or an -na annotation-level switch to allow the current behavior

Option 2: add -ja which means fail if the annotation is not parseable as JSON; this option leaves the default with our current relaxed approach to parsing the annotations

**Comment by sjfink** [comment]
In the name of simplicity, I'd vote for Option 1 ... simply fail if it's not valid JSON.   I wouldn't even add a --nonstrict flag.

Other opinions?

**Comment by jasonpet** [comment]
I found that packages are still being installed using python cli.  When I switch to use go cli the input is always a String instead of JSON (even for valid annotations with no syntax errors)

**Comment by starpit** [comment]
@jasonpet we should open a go cli defect, i think?

**Comment by jasonpet** [comment]
yeah, we definitely cannot merge this yet: https://github.com/openwhisk/openwhisk/pull/883.
It will break the UI.  I let James Dubee know already.
@dubeejw - do we have a defect for this already?  If not I can create one.

**Comment by rabbah** [comment]
No we don't yet.

**Comment by jasonpet** [comment]
Opened #952 for issue with annotation values using go cli.

**Comment by mdeuser** [comment]
For clarity regarding the format of the key string and value string, let's consider both of these valuea for the command `wsk action create test test.js -a k1 v1`

1. k1, the key.  This will eventually be folded into JSON prior to the REST endpoint call.  Currently, the user is not expected to provide k1 in valid JSON format, and the CLI will handle the conversion of the k1 string into a valid JSON string.  Conversion involves escaping embedded `/` and `"` characters, wrapping the entire value in double quotes.
2. v1, the value.  Stephen's comment above implies that this value must be a correctly formatted JSON object, array, string, number, `true`, `false`, or `null`.  The CLI should NOT perform any conversion of this value as it is expected to be valid JSON.

Note: CLI users must still provide appropriate escaping of the k1 and v1 values to ensure the shell's argument parsing does not corrupt the intended value.

http://www.json.org/

**Comment by jasonpet** [comment]
@csantanapr - #952 is the issue I opened and is the real blocker one.  Annotation values were not parsed properly at all in go-cli.  This issue is for addressing the scenario where a user has a syntax error in there annotation value.  Before we were not throwing an error and were creating an invalid value (we created the type as String instead of JSON).  The fix for this issue would be to throw an error for invalid syntax.  #952 has to go in before we can flip to use go-cli.  It can be verified by manually flipping the switch to go-cli.
